lib/browser/tag/update.js

```diff
@@ -160,6 +161,39 @@ function updateTagRef(expr, parent) {
   expr.tag.update()
 }
 
+/**
+ * Update dynamically created riot-tag with changing expressions
+ * @param   { Object } expr - expression tag and expression info
+ * @param   { Tag } parent - parent for tag creation
+ */
+
+function updateRtag(expr, parent) {
+  var tagName = tmpl(expr.value, parent),
+    conf
+
+  if (expr.tag && expr.tagName == tagName) {
+    expr.tag.update()
+    return
+  }
+
+  // sync _parent to accommodate changing tagnames
+  if (expr.tag) {
+    var delName = expr.tag.opts.riotTag,
+      tags = expr.tag._parent.tags[delName]
+
+    if (isArray(tags)) tags.splice(tags.indexOf(expr.tag), 1); else delete expr.tag._parent.tags[delName]
+
+  }
+
```

Comment:
Ok, thanks for the explanation.  Here is the issue

```html
<div each={tag in items}>
  <div riot-tag={tag}></div>
</div>
```
```javascript
this.items=['color', 'string', 'calendar']
```

then I later do `this.items[1] = 'calendar'` the `_parent` for this item has `_parent.tags: [string, calendar]` but it should only at this point have `[calendar]`, the `calendar` seems to be added fine in the each loop but it does not account for the removal of the existing `string`, this is what needs to be solved.

So I guess I just need to wait and see how the `_parent` gets resolved first.  If it goes away, then this can be removed and all should be well